src/StructuredLogViewer/Controls/BuildControl.xaml.cs

```diff
@@ -137,6 +141,7 @@ public BuildControl(Build build, string logFilePath)
             propertiesAndItemsControl.WatermarkDisplayed += UpdatePropertiesAndItemsWatermark;
             propertiesAndItemsControl.RecentItemsCategory = "PropertiesAndItems";
 
+            secretsSearch = new SecretsSearch(build);

```

Comment:
build.Analyze() will add a SecretsSearch to the build itself, perhaps no need to do it here? when you create new SecretsSearch() in the other location, save the instance to the Build:
https://github.com/YuliiaKovalova/MSBuildStructuredLog/blob/6e4fc91055cc354e57684649d10a03e0e826b072/src/StructuredLogViewer/MainWindow.xaml.cs#L695

and then you can access it off the Build.SecretsSearch

Reply:
basically make sure that the constructor is called in a single place